filestore/core.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import six
+from document import Document
+from jsonschema import validate as js_validate
+from bson import ObjectId
+
+
+class DatumNotFound(Exception):
+    pass
+
+
+def get_datum(col, eid, _DATUM_CACHE, get_spec_handler, logger):
```

Comment:
The point is to pull the core logic _off_ of that class so that it can be reused by the service.